components/dashboard/src/start/CreateWorkspace.tsx

```diff
@@ -545,23 +552,18 @@ function RunningPrebuildView(props: RunningPrebuildViewProps) {
     const { showUseLastSuccessfulPrebuild } = useContext(FeatureFlagContext);
 
     useEffect(() => {
-        const disposables = new DisposableCollection();
-
-        disposables.push(
-            getGitpodService().registerClient({
-                onInstanceUpdate: (update) => {
-                    if (update.workspaceId !== workspaceId) {
-                        return;
-                    }
-                    if (update.status.phase === "stopped") {
+        const timeout = setTimeout(() => {
+            getGitpodService()
+                .server.getWorkspace(workspaceId)
+                .then((ws) => {
+                    if (ws.latestInstance?.status?.phase === "stopped") {
```

Comment:
polling here every 2 secs is much simpler (and actually works). Listening to events has a problem where we could have missed the 'stopped' event and then hang forever

Reply:
Love this! I'm a huge fan of this simpler approach, and it also aligns much better with where we are looking to go with Pub API.

Reply:
I'm fine with either way. Can we use [repeat](https://github.com/gitpod-io/gitpod/blob/2f0b9462b0451f0974434fa36bd757420ef04e34/components/gitpod-protocol/src/util/repeat.ts#L17) or any other mechnaims that ensures we cannot pile up requests here? :cloud: